code/game/objects/items/storage/fancy.dm

```diff
@@ -137,7 +151,7 @@
 		return
 	var/obj/item/clothing/mask/cigarette/W = locate(/obj/item/clothing/mask/cigarette) in contents
 	if(W && contents.len > 0)
-		remove_from_storage(W, user)
+		SendSignal(COMSIG_TRY_STORAGE_TAKE, W, user)

```

Comment:
I'm kind of wary of a pattern like this, can't you not have remove_from_storage(W, user) and define remove_from_storage(W, user) as SendSignal(COMSIG_TRY_STORAGE_TAKE, W, user) like the atom stuff in
https://github.com/tgstation/tgstation/blob/ef0c1617cfead2df6bd734aa332cc0181e10f333/code/game/atoms.dm#L372-L373

Reply:
these wrapper procs are kind of noisy though